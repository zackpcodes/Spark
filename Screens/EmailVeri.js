import React, { Component } from 'react';
import { Text, View, TextInput, TouchableOpacity, Image, Alert} from 'react-native';
import { KeyboardAvoidingView } from 'react-native';

import './styles';
import './Globals';



// EmailVeri: Ask's users to enter their email and a request is sent
// to api on server for verification code.
export default class EmailVeri extends Component {
    constructor(props) {
      super(props);
      this.state = {
        email: '',
      };
    }
  

    validateEmail = (text) => {
      console.log(text);
      let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
      if (reg.test(text) === false) {
        this.setState({ email: text })
        return false;
      }
      else {
        this.setState({ email: text })
        return true;
      }
    }
  
  
    emailVeriRequest = () => {
      global.email_phone = this.state.email
      if (this.validateEmail(this.state.email)) {
        fetch('http://spark.pemery.co/account/dispatch', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            email: this.state.email,
          }),
        }).then((response) => response.json())
          .then((responseJson) => {
            if (responseJson.status == 200){
              this.props.navigation.replace('Veri');
            }else{
              Alert.alert('Error', responseJson.content.comment);
            }
            
          })
          .catch((error) => {
            console.log(error)
            Alert.alert('Error', 'Promise rejection error')
          });
        
      } else {
        alert("Please enter a valid email address")

      } 
    }

  
    render() {
      return (
  
          <View style={stylesForReg.container}>
            <Image
              style={{ width: 210, height: 210, marginLeft: 20, }}
              source={require('./MainLogo.png')}
            />
            <Text style={{ marginBottom: 160, fontSize: 55, textAlign: 'center', fontFamily: 'sans-serif-thin', }}>SPARK</Text>
            <TextInput
              style={stylesForReg.textIn}
              placeholder='Enter Your Email'
              onChangeText={(email) => this.setState({ email })}
              value={this.state.email}
            />
  
            <TouchableOpacity style={stylesForReg.touchButton} onPress={() => this.emailVeriRequest()}>
              <Text style={{ fontSize: 19, padding: 17 }}>Continue</Text>
            </TouchableOpacity>
          </View>
      );
    }
  }