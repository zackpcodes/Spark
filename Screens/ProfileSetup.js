import React, { Component } from 'react';
import { Text, View, TextInput, TouchableOpacity, Image} from 'react-native';
import { KeyboardAvoidingView } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import Constants from 'expo-constants';

import './styles';
import './Globals';


// ProfileSetup: Allows the user to set up a profile then makes 
// a call to the server to push profile data.
export default class ProfileSetup extends Component {
    constructor(props) {
      super(props);
      this.state = {
        image: userProfilepic,
        organizationText: '',
        usernameText: '',
        placeHolderOrganization: 'Organization',
        placeHolderUsername: "Username",
        btnText: 'Done'
      };
    }

  
    profilePush = () => {
      fetch('http://104.196.33.197:80/account/modify', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            name: this.state.usernameText,
          }),
        }).then((response) => response.json())
          .then((responseJson) => {
            console.log(responseJson)
            if (responseJson.status == 200){
              this.props.navigation.navigate('Convos')
            }else{
              Alert.alert('Invalid', 'Please login.')
            }
            
          })
          .catch((error) => {
            console.error(error);
            Alert.alert('Error', 'A connection error has occurred!')
          });
    }
  
    render() {
      let { image } = this.state;
      return (
          <View style={stylesForReg.container}>
  
            <TouchableOpacity onPress={this._pickImage}>
              <Image source={{ uri: image }} style={stylesForReg.profilePicSelection} />
            </TouchableOpacity>
  
            <TextInput
              style={stylesForReg.textIn}
              placeholder={this.state.placeHolderOrganization}
              onChangeText={(organizationText) => this.setState({ organizationText })}
              value={this.state.organizationText}
            />
            <TextInput
              style={stylesForReg.textIn}
              placeholder={this.state.placeHolderUsername}
              onChangeText={(usernameText) => this.setState({ usernameText })}
              value={this.state.veriCode}
            />
  
            <TouchableOpacity style={stylesForReg.touchButton} onPress={() => this.profilePush()}>
              <Text style={{ fontSize: 19, padding: 17 }}>{this.state.btnText}</Text>
            </TouchableOpacity>
          </View>
      );
    }
  
    componentDidMount() {
      this.getPermissionAsync();
    }
  
    getPermissionAsync = async () => {
      if (Constants.platform.ios) {
        const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
        if (status !== 'granted') {
          alert('Sorry, we need camera roll permissions to make this work!');
        }
      }
    }
  
    _pickImage = async () => {
      let result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.All,
        allowsEditing: true,
        aspect: [4, 3],
        quality: 1
      });
  
      console.log(result);
  
      if (!result.cancelled) {
        this.setState({ image: result.uri });
        userProfilepic = result.uri
      }
    };
  }