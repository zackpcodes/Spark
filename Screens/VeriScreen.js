import React, { Component } from 'react';
import { Text, View,TextInput, TouchableOpacity, Image, Alert} from 'react-native';
import { KeyboardAvoidingView } from 'react-native';

import './styles';

// VeriScreen: Ask's user to enter verification code from api, then
// sends this entered code back to server via api call for verification.
export default class VeriScreen extends Component {

    constructor(props) {
      super(props);
      this.state = {
        email: global.email_phone,
        veriCode: '',
        placeHolderText: "Enter Verification Code",
        btnText: 'Confirm'
      };
    }
  
  
    codeVeriRequest = () => {

        fetch('http://spark.pemery.co/account/verify', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            email_phone: this.state.email,
            code: this.state.veriCode,
          }),
        }).then((response) => response.json())
          .then((responseJson) => {
            console.log(responseJson)
            if (responseJson.status == 200){
              this.props.navigation.replace('ProfileSetup')
            }else{
              Alert.alert('Invalid', 'Invalid verification code, please check the code and try again.')
            }
            
          })
          .catch((error) => {
            console.error(error);
            Alert.alert('Error', 'A connection error has occurred!')
          });
  
    }
  
    render() {
      return (
          <View style={stylesForReg.container}>
            <Image
              style={{ width: 210, height: 210, marginLeft: 20, }}
              source={require('./MainLogo.png')}
            />
  
            <Text style={{ marginBottom: 160, fontSize: 55, textAlign: 'center', fontFamily: 'sans-serif-thin', }}>SPARK</Text>
  
            <TextInput
              style={stylesForReg.textIn}
              placeholder={this.state.placeHolderText}
              onChangeText={(veriCode) => this.setState({ veriCode })}
              value={this.state.veriCode}
            />
  
            <TouchableOpacity style={stylesForReg.touchButton} onPress={() => this.codeVeriRequest()}>
              <Text style={{ fontSize: 19, padding: 17 }}>{this.state.btnText}</Text>
            </TouchableOpacity>
          </View>
      );
    }
  }