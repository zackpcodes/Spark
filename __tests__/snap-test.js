import React from 'react';
import Intro from '../Screens/styles';

import renderer from 'react-test-renderer';
import ProfileSetup from '../Screens/ProfileSetup';
import EmailVeri from '../Screens/EmailVeri';
import VeriScreen from '../Screens/VeriScreen';

test('renders correctly', () => {
  const treeProfile = renderer.create(<ProfileSetup />).toJSON();
  expect(treeProfile).toMatchSnapshot();

});

test('renders correctly', () => {
  const treeProfile = renderer.create(<EmailVeri />).toJSON();
  expect(treeProfile).toMatchSnapshot();

});