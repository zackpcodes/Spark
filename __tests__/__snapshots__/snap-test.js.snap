// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dcdcdd",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": undefined,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": null,
        }
      }
      style={
        Object {
          "borderColor": "#1985a1",
          "borderRadius": 100,
          "borderWidth": 3,
          "height": 200,
          "marginBottom": 160,
          "width": 200,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Organization"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#1985a1",
        "borderRadius": 2,
        "borderWidth": 0.7,
        "fontSize": 17,
        "height": 60,
        "marginTop": 10,
        "padding": 10,
        "textAlign": "center",
        "width": 290,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#1985a1",
        "borderRadius": 2,
        "borderWidth": 0.7,
        "fontSize": 17,
        "height": 60,
        "marginTop": 10,
        "padding": 10,
        "textAlign": "center",
        "width": 290,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1985a1",
        "borderRadius": 2,
        "height": 65,
        "marginBottom": 50,
        "marginTop": 30,
        "opacity": 0.7,
        "width": 290,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 19,
          "padding": 17,
        }
      }
    >
      Done
    </Text>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dcdcdd",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": undefined,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 210,
        "marginLeft": 20,
        "width": 210,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "sans-serif-thin",
        "fontSize": 55,
        "marginBottom": 160,
        "textAlign": "center",
      }
    }
  >
    SPARK
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter your email"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#1985a1",
        "borderRadius": 2,
        "borderWidth": 0.7,
        "fontSize": 17,
        "height": 60,
        "marginTop": 10,
        "padding": 10,
        "textAlign": "center",
        "width": 290,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1985a1",
        "borderRadius": 2,
        "height": 65,
        "marginBottom": 50,
        "marginTop": 30,
        "opacity": 0.7,
        "width": 290,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 19,
          "padding": 17,
        }
      }
    >
      Continue
    </Text>
  </View>
</View>
`;
